tests/server/test_parameters_api.py

```diff
@@ -258,3 +258,101 @@ async def test_get_set_param(environment, client, server):
 
     result = await client.delete_param(tid=environment, id="key10")
     assert result.code == 200
+
+
+@pytest.mark.parametrize("no_agent", [True])
+async def test_dont_renew_old_facts(server, client, environment, clienthelper, caplog, time_machine, monkeypatch):
+    """
+    Make sure that we don't renew facts that belong to a resource that is no longer present in the latest released
+    model version.
+    """
+    server_fact_renew_time = config.server_fact_renew.get()
+    parameter_slice = server.get_slice(SLICE_PARAM)
+    agent_manager_slice = server.get_slice(SLICE_AGENT_MANAGER)
+
+    async def request_parameter_mock(env_id, resource_id):
+        return 200, {}
+
+    monkeypatch.setattr(agent_manager_slice, "request_parameter", request_parameter_mock)
+
+    resource_id1 = "std::testing::NullResource[vm1.dev.inmanta.com,name=test1]"
+    resource_id2 = "std::testing::NullResource[vm1.dev.inmanta.com,name=test2]"
+    version = await clienthelper.get_version()
+    resources = [
+        {
+            "id": f"{resource_id1},v={version}",
+            "name": "test1",
+            "param": "val1",
```

Comment:
What is this "param" field ? Is this just an arbitrary field ?

Reply:
It's a resource so it can have an arbitrary number of attributes. It doesn't serve a purpose in this test case. Should I remove it?

Reply:
> It doesn't serve a purpose in this test case. Should I remove it?

No it's fine either way

> It's a resource so it can have an arbitrary number of attributes

Is this True for all resources or only for this `testing::NullResource` ?
I'm just curious/surprised and I might be missing something obvious, but I would expect resources to have a well-defined schema.

Reply:
Ah. Right. You are pointing to the fact that the attributes of the resource are not in-line with those defined in the resource definition. That's something I should fix. I didn't realize we also have a handler for that resource.